test/firtool/legal-uninferred-resets.fir

```diff
@@ -0,0 +1,31 @@
+; RUN: firtool %s --verify-diagnostics --split-input-file
```

Comment:
Please also add a test that demonstrates the correctness of your new pass and not just the firrtl lowering as a whole.

Reply:
The test of uninferred reset check has been included in infer-resets-errors.mlir, the test which we actually need to add is to compare the errors emitted with or without enabling firrtl-inliner.

Reply:
I'm inclined to preserve this test using firtool since it is extracted from a actual use case, while adding a new minimal test. What do you think.